src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -841,27 +840,31 @@ protected void waitForPodStatus(String podId, String namespace, String slaveName
                     Map<String, Integer> errors = terminatedContainers.stream().collect(Collectors.toMap(
                             ContainerStatus::getName, (info) -> info.getState().getTerminated().getExitCode()));
                     // Print the last lines of failed containers
-                    logLastLines(terminatedContainers, podId, namespace, slaveName, errors);
+                    logLastLines(terminatedContainers, podId, namespace, errors);
                     throw new IllegalStateException("Containers are terminated with exit codes: " + errors);
                 }
 
                 if (!allContainersAreReady) {
                     continue;
                 }
 
-                String podStatus = getPodStatus(pod);
+                String podStatus = getPodStatus(pod, false);
                 LOGGER.log(Level.INFO, "Pod status is {0}", podStatus);
                 if (validStates.contains(podStatus)) {
                     inValidState = true;
                     break;
                 }
             }
             if (!inValidState) {
-                throw new IllegalStateException(format("Container is not running after %d attempts, status: %s", startChecks, getPodStatus(pod)));
+                throw new IllegalStateException(format("Container is not running after %d attempts, status: %s",
+                        startChecks, getPodStatus(pod, false)));
             }
         }
 
-        protected String getPodStatus(Pod pod) {
+        protected String getPodStatus(Pod pod, boolean refresh) {
+            if (refresh) {
```

Comment:
Since it looks like there is only one place in code, which passes `refresh == true`, I suggest reverting this chunk of code in favor of refreshing `pod` at the caller's site, as flag arguments are usually considered as a bad code smell.

By the way, is this method used in `KubernetesCloud`'s children? If not, it could be just package-private instead of `protected` (if `Pod` wasn't a foreign class, it would make sense moving this whole method there...).